lib/portage/dbapi/bintree.py

```diff
@@ -672,14 +765,21 @@ def _populate_local(self, reindex=True):
 				pkgindex = self._new_pkgindex()
 			metadata = {}
 			basename_index = {}
+			binpkg_format_default = self.settings.get("BINPKG_FORMAT", "xpak")
```

Comment:
This default should instead be set in `_pkgindex_default_header_data`, like this:
```diff
diff --git a/lib/portage/dbapi/bintree.py b/lib/portage/dbapi/bintree.py
index 2ce18bd6f..231eef86d 100644
--- a/lib/portage/dbapi/bintree.py
+++ b/lib/portage/dbapi/bintree.py
@@ -485,6 +485,7 @@ class binarytree:
 
 			# Populate the header with appropriate defaults.
 			self._pkgindex_default_header_data = {
+				"BINPKG_FORMAT": self.settings.get("BINPKG_FORMAT", ""),
 				"CHOST"        : self.settings.get("CHOST", ""),
 				"repository"   : "",
 			}

```
From there it will propagate down as needed.